File: sort/Nim/sort.nim
import os
import algorithm

const v_size: int = 4194304
var v: array[v_size, int]
var seed: int = 3
var i: int = 0
while i < v_size:
    v[i] = seed
    seed = seed * 27487
    seed = seed mod 30491
    i = i + 1
algorithm.sort(v, system.cmp[int])
if paramCount() > 0:
    echo v[279121]

File: sort/Haskell/sort.hs
import Control.DeepSeq
import Control.Monad
import Data.List
import System.Environment

(n, seed, ml, md, nprint) = (4194304, 3, 27487, 30491, 279121)

foo :: [Int]
foo = sort . take n $ iterate ((`mod` md) . (* ml)) seed

main = foo `deepseq` getArgs >>= (flip when) (print $ foo !! nprint) . not . null
